script.js

```diff
@@ -1,4 +1,81 @@
+// Create game states
+var playerToRollDice = "rollDice";
+var playerChooseDiceOrder = "chooseDiceOrder";
+
+var player2ToRollDice = "rollDice2";
+var player2ChooseDiceOrder = "chooseDiceOrder2";
+
+var compareScores = "compareScores";
+
+// Default game state
+var currentGameMode = playerToRollDice;
+
+var player1DiceRoll1, player1DiceRoll2;
+var player2DiceRoll1, player2DiceRoll2;
+
+var player1DiceOrder, player2DiceOrder;
+
+var randomRollDice = function () {
+  var randomDecimal = Math.random() * 6;
+  var randomInteger = Math.floor(randomDecimal);
+  var diceNumber = randomInteger + 1;
+  return diceNumber;
+};
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+  // player 1 roll dice
+  if (currentGameMode == playerToRollDice) {
+    player1DiceRoll1 = randomRollDice();
+    player1DiceRoll2 = randomRollDice();
+    currentGameMode = playerChooseDiceOrder;
+    return `Player 1, here's what you rolled! 🎲<br><br> Dice 1: ${player1DiceRoll1} <br> Dice 2: ${player1DiceRoll2} <br><br> Choose your dice order by entering '1' or '2' in the input field.`;
+  }
+
+  // player 1 choose dice order
+  if (currentGameMode == playerChooseDiceOrder) {
+    if (input == "1") {
+      player1DiceOrder = "" + player1DiceRoll1 + player1DiceRoll2;
+    } else {
+      player1DiceOrder = "" + player1DiceRoll2 + player1DiceRoll1;
+    }
```

Comment:
This is a novel way of combining the numbers! adding a string to a number would cast that number as a string as opposed to a number, hence why this works.

Also, it might be best to add an input validation somewhere here as this would mean that if you'd have an input other than `1` it will always default to the second choice. i.e. if let's say my input is `3` it'll go to the else block, even when my input should not be a valid input like say `"what's up"` it will still go to the else block.